### PR TITLE
Add option to use dead modules information (disabled by default)

### PR DESCRIPTION
**Pull request by mmasciov**
#### PR description:

Add option to use dead modules information (disabled by default).
Reflects PR #332 of trackreco/mkFit

#### PR validation:
http://uaf-10.t2.ucsd.edu/~mmasciov/MkFit_devs/testPR37/
-> As expected.

### CODE REVIEW

File: RecoTracker/MkFit/plugins/MkFitEventOfHitsProducer.cc [hunk 1]
```diff
@@ -54,13 +55,15 @@ MkFitEventOfHitsProducer::MkFitEventOfHitsProducer(edm::ParameterSet const& iCon
...
-      putToken_{produces<MkFitEventOfHits>()} {}
+      putToken_{produces<MkFitEventOfHits>()},
+      useDeadModules_{iConfig.getParameter<bool>("useDeadModules")} {}
 
 void MkFitEventOfHitsProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
   edm::ParameterSetDescription desc;
 
   desc.add("pixelHits", edm::InputTag{"mkFitSiPixelHits"});
   desc.add("stripHits", edm::InputTag{"mkFitSiStripHits"});
+  desc.add("useDeadModules", false)->setComment("Use dead modules information");
```
Comment (makortel) [inline RecoTracker/MkFit/plugins/MkFitEventOfHitsProducer.cc:66]: `MeasurementTracker` (which plays somewhat similar role to `mkfit::EventOfHits` appears to use "bad" nomenclature
https://github.com/trackreco/cmssw/blob/601fbc54e314860dd9f9fcae3569794388e3c306/RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc#L135-L139
and specifically `UseStripStripQualityDB`
https://github.com/trackreco/cmssw/blob/601fbc54e314860dd9f9fcae3569794388e3c306/RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc#L91-L96
to control whether the `SiStripQuality` is used or not. I'd suggest to use same or similar nomenclature here.

Comment (mmasciov) [reply RecoTracker/MkFit/plugins/MkFitEventOfHitsProducer.cc]: 'useDeadModules' has been replaced with 'useStripStripQualityDB', for consistency with existing nomenclature.

File: RecoTracker/MkFit/plugins/MkFitEventOfHitsProducer.cc [hunk 2]
```diff
@@ -54,13 +55,15 @@ MkFitEventOfHitsProducer::MkFitEventOfHitsProducer(edm::ParameterSet const& iCon
       mkFitGeomToken_{esConsumes()},
       qualityToken_{esConsumes()},
       geomToken_{esConsumes()},
-      putToken_{produces<MkFitEventOfHits>()} {}
+      putToken_{produces<MkFitEventOfHits>()},
+      useStripStripQualityDB_{iConfig.getParameter<bool>("useStripStripQualityDB")} {}
```
Comment (makortel) [inline RecoTracker/MkFit/plugins/MkFitEventOfHitsProducer.cc:59]: Now that the `SiStripQuality` and `TrackerGeometry` are used only if `useStripStripQualityDB_ == true`, their consumes declaration should be changed to
```cpp
if (useStripStripQualityDB_) {
  qualityToken_ = esConsumes();
  geomToken_ = esConsumes();
}
```

Comment (mmasciov) [reply RecoTracker/MkFit/plugins/MkFitEventOfHitsProducer.cc]: Done. Please, check.

**Pull request merged**